In [24]:
import numpy as np
import pandas as pd

In [25]:
#Sample labels
labelsamples= ["b1p1_m05","b1p1_m10","b1p1_m20","b1p1_m40","b1p5_m05","b1p5_m10","b1p5_m20","b1p5_m40","b10_m05","b10_m10","b10_m20","b10_m40"]

path='../CLs/'
#txt files with signal strength, expected background and signal efficiency for each nuclear configuration model
infiles = ['Eff_Bkg_index_00a.txt', 'Eff_Bkg_index_01a.txt', 'Eff_Bkg_index_02a.txt', 'Eff_Bkg_index_00b.txt', 'Eff_Bkg_index_01b.txt', 'Eff_Bkg_index_02b.txt']

#function to read the list of txt files
def readtxt(infile_array):
    listloaded = [] 
    for infile in infile_array:
        listloaded.append(np.loadtxt(path+infile, usecols=(3,4), skiprows=1))
    return listloaded

In [26]:
optimals = readtxt(infiles) 

## 1: (b_cv - b_i)/b_cv ,   2: (eff_cv - eff_i)/eff_cv ##

In [27]:
def return_shift(value):
    if value < 1:
        return -1+value
    if value > 1:
        return value-1
    if value == 1:
        return value-1

In [28]:
optimals = np.array(optimals)
optimals

array([[[1.500000e-02, 2.431368e+03],
        [1.800000e-02, 2.434771e+03],
        [1.800000e-02, 2.434771e+03],
        [1.900000e-02, 2.433313e+03],
        [1.130000e-01, 1.676655e+03],
        [1.150000e-01, 1.676655e+03],
        [1.180000e-01, 1.676655e+03],
        [1.180000e-01, 1.676655e+03],
        [3.730000e-01, 2.590332e+03],
        [3.870000e-01, 2.619257e+03],
        [4.020000e-01, 1.647730e+03],
        [3.320000e-01, 1.811312e+03]],

       [[1.400000e-02, 2.502410e+03],
        [1.600000e-02, 2.507192e+03],
        [1.800000e-02, 2.507192e+03],
        [1.800000e-02, 2.504860e+03],
        [1.160000e-01, 1.732361e+03],
        [1.170000e-01, 1.732361e+03],
        [1.200000e-01, 1.732361e+03],
        [1.200000e-01, 1.732361e+03],
        [3.730000e-01, 2.703055e+03],
        [3.860000e-01, 2.735665e+03],
        [0.000000e+00, 0.000000e+00],
        [3.160000e-01, 1.704810e+03]],

       [[2.300000e-02, 2.702546e+03],
        [2.600000e-02, 2.705910e+03],
        

In [29]:
# Vectorize the function
vectorized_nuclear_shift = np.vectorize(return_shift)

In [30]:
def color_negative_red(val):
    """
    Takes a scalar and returns a string with
    the css property `'color: red'` for negative
    strings, black otherwise.
    """
    color = 'blue' if val > 0 else 'black'
    return 'color: % s' % color

models = [vectorized_nuclear_shift(optimals[i][:][:]/optimals[0][:][:]) for i in range(6) ] 
df_models = [pd.DataFrame({'Efficiency Shift': model[:, 0]*100, 'Background Shift': model[:, 1]*100}) for model in models]


    
    
    

In [31]:
df_models[2]

Efficiency Shift  Background Shift
0          53.333333         11.153310
1          44.444444         11.136119
2          55.555556         11.136119
3          47.368421         11.120641
4          -0.884956         10.428919
5           1.739130         10.428919
6           1.694915         10.428919
7           3.389831         10.428919
8           0.268097          7.396272
9           1.033592          7.293977
10        -21.890547          5.610628
11          0.602410          6.409939

In [33]:
df_models[5].iloc[:,0]

0     120.000000
1      94.444444
2      88.888889
3      78.947368
4       2.654867
5       2.608696
6       5.932203
7       5.084746
8      -2.680965
9      -1.033592
10    -24.129353
11     -1.807229
Name: Efficiency Shift, dtype: float64

In [15]:
df_models[0].to_latex(index=False)

/home/leoperes/.local/lib/python3.12/site-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.
  warnings.warn("Unable to import Axes3D. This may be due to multiple versions of "


'\\begin{tabular}{rr}\n\\toprule\nEfficiency Shift & Background Shift \\\\\n\\midrule\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n0.000000 & 0.000000 \\\\\n\\bottomrule\n\\end{tabular}\n'

In [23]:
print(df_models[5].to_latex(index=False))

\begin{tabular}{rr}
\toprule
Efficiency Shift & Background Shift \\
\midrule
120.000000 & 19.380818 \\
94.444444 & 19.370980 \\
88.888889 & 19.370980 \\
78.947368 & 19.357600 \\
2.654867 & 16.388703 \\
2.608696 & 16.388703 \\
5.932203 & 16.388703 \\
5.084746 & 16.388703 \\
-2.680965 & 12.369804 \\
-1.033592 & 12.259240 \\
-24.129353 & 11.225686 \\
-1.807229 & 11.584365 \\
\bottomrule
\end{tabular}



&$m_\chi$ (GeV)}& hA\_BR & hA\_LFG & hA\_ESF  & hN\_BR  & hN\_LFG  & hN\_ESF  \\
\hline
\multirow{4}{*}{$\gamma = 1.1$}& 5 &0.015 & -6.6\%  & +53\%  &  &  & \\
  &10& 0.018&-11.1\%& +44\%  &  &  &   \\
  &20&0.018 &0\%  & +54\% & +55\%  &  &   \\
  &40&0.019 & -5.2\%&  &  & +47% &   \\
\hline
\multirow{4}{*}{$\gamma = 1.5$}& 5 &0.113  &+2.6\%  &  &  &  & \\
  &10& 0.115& 1.73\% &  &  &  &   \\
  &20& 0.118& 1.69\% &  &  &  &   \\
  &40& 0.118& 1.69\%  &  &  &  &   \\
  \hline
\multirow{4}{*}{$\gamma = 10$}& 5 & 0.373 &0.0\%  &  &  &  & \\
  &10&0.387 & -0.25\%&  &  &  &   \\
  &20& 0.402& X &  &  &  &   \\
  &40& 0.332& -4.8\%&  &  &  &   \\
\hline


53.333333
44.444444
55.555556
47.368421
-0.884956
1.739130
1.694915
3.389831
0.268097
1.033592
-21.890547
0.602410

 & 11.153310 \\
 & 11.136119 \\
 & 11.136119 \\
 & 11.120641 \\
 & 10.428919 \\
 & 10.428919 \\
 & 10.428919 \\
 & 10.428919 \\
 & 7.396272 \\
 & 7.293977 \\
 & 5.610628 \\
 & 6.409939 \\
